### PR TITLE
Added tests around custom, RH ostree repo, sync and cv publish, promote via api

### PR DESCRIPTION
**Pull request by sghai**
Test results:
```console
 robottelo]$ py.test -v tests/foreman/api/test_contentview.py::OstreeContentViewTestCase
================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
plugins: cov-2.3.1, xdist-1.15.0
collected 4 items 

tests/foreman/api/test_contentview.py::OstreeContentViewTestCase::test_positive_add_custom_ostree_content <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_contentview.py::OstreeContentViewTestCase::test_positive_promote_custom_ostree PASSED
tests/foreman/api/test_contentview.py::OstreeContentViewTestCase::test_positive_publish_custom_ostree <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_contentview.py::OstreeContentViewTestCase::test_positive_publish_promote_with_custom_ostree_and_other PASSED

============================================================= 4 passed in 195.66 seconds =============================================================
[robottelo]$ py.test -v tests/foreman/api/test_contentview.py::ContentViewRedHatOstreeContent
================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
plugins: cov-2.3.1, xdist-1.15.0
collected 4 items 

tests/foreman/api/test_contentview.py::ContentViewRedHatOstreeContent::test_positive_add_rh_ostree_content PASSED
tests/foreman/api/test_contentview.py::ContentViewRedHatOstreeContent::test_positive_promote_RH_ostree PASSED
tests/foreman/api/test_contentview.py::ContentViewRedHatOstreeContent::test_positive_publish_RH_ostree PASSED
tests/foreman/api/test_contentview.py::ContentViewRedHatOstreeContent::test_positive_publish_promote_with_RH_ostree_and_other PASSED

============================================================= 4 passed in 327.78 seconds ========================================================

robottelo]$ py.test -v tests/foreman/api/test_repository.py::OstreeRepositoryTestCase
================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
plugins: cov-2.3.1, xdist-1.15.0
collected 5 items 

tests/foreman/api/test_repository.py::OstreeRepositoryTestCase::test_positive_create_ostree <- robottelo/decorators/host.py PASSED
tests/foreman/api/test_repository.py::OstreeRepositoryTestCase::test_positive_delete_ostree <- robottelo/decorators/host.py PASSED
tests/foreman/api/test_repository.py::OstreeRepositoryTestCase::test_positive_sync_rh_atomic <- robottelo/decorators/host.py PASSED
tests/foreman/api/test_repository.py::OstreeRepositoryTestCase::test_positive_update_name <- robottelo/decorators/host.py PASSED
tests/foreman/api/test_repository.py::OstreeRepositoryTestCase::test_positive_update_url <- robottelo/decorators/host.py PASSED

============================================================= 5 passed in 201.08 seconds =============================================
```

### CODE REVIEW

File: tests/foreman/api/test_repository.py [hunk 1]
```diff
@@ -746,3 +749,111 @@ def test_positive_update_name(self):
...
+            url=FEDORA23_OSTREE_REPO,
+            unprotected=False
+        ).create()
+        repo.delete()
+        with self.assertRaises(HTTPError):
+            repo.read()
+
+    @tier2
+    @skip_if_not_set('fake_manifest')
+    def test_positive_sync_rh_atomic(self):
```
Comment (rochacbruno) [inline tests/foreman/api/test_repository.py]: manifest upload tests shouldn't use `run_in_one_thread` ?

**Comment by rochacbruno** [comment]
ACK (pending non-blocking question)

**Comment by renzon** [comment]
ACK. I have not merged because of @rochacbruno's comment

File: tests/foreman/api/test_repository.py (hunk 1 above)
Comment (sghai) [inline tests/foreman/api/test_repository.py]: I missed that. thank you for pointing that.. PR is updated now.

**Pull request merged**